lib/msal-common/src/cache/CacheHelpers.ts

```diff
@@ -76,12 +76,15 @@ export class CacheHelpers {
     updateCacheEntries(serverAuthenticationRequest: ServerCodeRequestParameters, account: Account): void {
         // Cache account and state
         if (account) {
-            this.setAccountCache(account, serverAuthenticationRequest.state);
+            this.setAccountCache(account);
         }
 
         // Cache the request state
         this.cacheStorage.setItem(TemporaryCacheKeys.REQUEST_STATE, serverAuthenticationRequest.state);
 
+        // Cache the nonce
+        this.cacheStorage.setItem(`${TemporaryCacheKeys.NONCE_IDTOKEN}|${serverAuthenticationRequest.state}`, serverAuthenticationRequest.nonce);
```

Comment:
should this key have a generator function as well?

Reply:
That may be useful, I will add that.

Reply:
done.